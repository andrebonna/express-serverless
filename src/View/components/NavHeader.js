import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { DropdownButton, MenuItem } from 'react-bootstrap';

import './nav-header.scss';

export default class NavHeader extends Component {

    constructor(props) {
        super(props);
        this.menuItemClickAction = this.menuItemClickAction.bind(this);
    }

    menuItemClickAction(href) {
        return () => window.location.href = href;
    }

    renderCategories() {
        const { categories } = this.props;

        return categories.map((category, i)=>{
            return <MenuItem key={category.name} eventKey={i} onClick={this.menuItemClickAction(category.href)}>{category.name}</MenuItem>;
        });
    }

    render() {
        return (
            <nav className='navbar navbar-default navbar-fixed-top'>
                <div className='container-fluid'>
                    <div className='navbar-header'>
                        <div className="navbar-toggle borderless collapsed">
                            <DropdownButton pullRight bsStyle="default" id='hamburguer-menu' title='' noCaret className='glyphicon glyphicon-menu-hamburger' >
                                <MenuItem header>Categories</MenuItem>
                                {this.renderCategories()}
                                <MenuItem divider />
                                <MenuItem eventKey="4" onClick={this.menuItemClickAction('contact')}>Contact Us</MenuItem>
                            </DropdownButton>
                        </div> 
                        <a className='navbar-brand' href='home'>Universal React</a>
                    </div>
                    <div className='navbar-collapse collapse'>
                        <ul className='nav navbar-nav navbar-right'>
                            <li>
                                <DropdownButton id='categories' title='Categories' noCaret className='categories-dropdown' >
                                    {this.renderCategories()}
                                </DropdownButton>
                            </li>
                            <li><a href='contact'>Contact Us</a></li>
                        </ul>
                    </div>
                </div>
            </nav>
        );
    }
}

NavHeader.propTypes = {
    categories: PropTypes.arrayOf(PropTypes.shape({
        name: PropTypes.string,
        href: PropTypes.string
    })).isRequired
};