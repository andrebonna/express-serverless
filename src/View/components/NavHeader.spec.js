import React from 'react';
import { shallow } from 'enzyme';
import NavHeader from './NavHeader';

describe('NavHeader', ()=>{
    it('render successfully', ()=>{
        const props = {
            categories: [{
                name: 'nameMock',
                href: 'hrefMock'
            }]
        };
        const wrapper = shallow(<NavHeader {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('menuItemClickAction', ()=>{
        const props = {
            categories: [{
                name: 'nameMock',
                href: 'hrefMock'
            }]
        };
        const wrapper = shallow(<NavHeader {...props} />);
        wrapper.find({ eventKey: '4' }).simulate('click');
        expect(window.location.href).toBe('contact');
    });
});