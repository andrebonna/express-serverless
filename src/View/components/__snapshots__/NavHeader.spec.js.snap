exports[`NavHeader render successfully 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <nav
    className="navbar navbar-default navbar-fixed-top">
    <div
        className="container-fluid">
        <div
            className="navbar-header">
            <div
                className="navbar-toggle borderless collapsed">
                <DropdownButton
                    bsStyle="default"
                    className="glyphicon glyphicon-menu-hamburger"
                    id="hamburguer-menu"
                    noCaret={true}
                    pullRight={true}
                    title="">
                    <MenuItem
                        bsClass="dropdown"
                        disabled={false}
                        divider={false}
                        header={true}>
                        Categories
                    </MenuItem>
                    <MenuItem
                        bsClass="dropdown"
                        disabled={false}
                        divider={false}
                        eventKey={0}
                        header={false}
                        onClick={[Function]}>
                        nameMock
                    </MenuItem>
                    <MenuItem
                        bsClass="dropdown"
                        disabled={false}
                        divider={true}
                        header={false} />
                    <MenuItem
                        bsClass="dropdown"
                        disabled={false}
                        divider={false}
                        eventKey="4"
                        header={false}
                        onClick={[Function]}>
                        Contact Us
                    </MenuItem>
                </DropdownButton>
            </div>
            <a
                className="navbar-brand"
                href="home">
                Universal React
            </a>
        </div>
        <div
            className="navbar-collapse collapse">
            <ul
                className="nav navbar-nav navbar-right">
                <li>
                    <DropdownButton
                        className="categories-dropdown"
                        id="categories"
                        noCaret={true}
                        title="Categories">
                        <MenuItem
                            bsClass="dropdown"
                            disabled={false}
                            divider={false}
                            eventKey={0}
                            header={false}
                            onClick={[Function]}>
                            nameMock
                        </MenuItem>
                    </DropdownButton>
                </li>
                <li>
                    <a
                        href="contact">
                        Contact Us
                    </a>
                </li>
            </ul>
        </div>
    </div>
</nav>,
  "nodes": Array [
    <nav
      className="navbar navbar-default navbar-fixed-top">
      <div
            className="container-fluid">
            <div
                  className="navbar-header">
                  <div
                        className="navbar-toggle borderless collapsed">
                        <DropdownButton
                              bsStyle="default"
                              className="glyphicon glyphicon-menu-hamburger"
                              id="hamburguer-menu"
                              noCaret={true}
                              pullRight={true}
                              title="">
                              <MenuItem
                                    bsClass="dropdown"
                                    disabled={false}
                                    divider={false}
                                    header={true}>
                                    Categories
                              </MenuItem>
                              <MenuItem
                                    bsClass="dropdown"
                                    disabled={false}
                                    divider={false}
                                    eventKey={0}
                                    header={false}
                                    onClick={[Function]}>
                                    nameMock
                              </MenuItem>
                              <MenuItem
                                    bsClass="dropdown"
                                    disabled={false}
                                    divider={true}
                                    header={false} />
                              <MenuItem
                                    bsClass="dropdown"
                                    disabled={false}
                                    divider={false}
                                    eventKey="4"
                                    header={false}
                                    onClick={[Function]}>
                                    Contact Us
                              </MenuItem>
                        </DropdownButton>
                  </div>
                  <a
                        className="navbar-brand"
                        href="home">
                        Universal React
                  </a>
            </div>
            <div
                  className="navbar-collapse collapse">
                  <ul
                        className="nav navbar-nav navbar-right">
                        <li>
                              <DropdownButton
                                    className="categories-dropdown"
                                    id="categories"
                                    noCaret={true}
                                    title="Categories">
                                    <MenuItem
                                          bsClass="dropdown"
                                          disabled={false}
                                          divider={false}
                                          eventKey={0}
                                          header={false}
                                          onClick={[Function]}>
                                          nameMock
                                    </MenuItem>
                              </DropdownButton>
                        </li>
                        <li>
                              <a
                                    href="contact">
                                    Contact Us
                              </a>
                        </li>
                  </ul>
            </div>
      </div>
</nav>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <NavHeader
        categories={
                Array [
                        Object {
                          "href": "hrefMock",
                          "name": "nameMock",
                        },
                      ]
        } />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": NavHeader {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "menuItemClickAction": [Function],
        "props": Object {
          "categories": Array [
            Object {
              "href": "hrefMock",
              "name": "nameMock",
            },
          ],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <nav
          className="navbar navbar-default navbar-fixed-top">
          <div
                    className="container-fluid">
                    <div
                              className="navbar-header">
                              <div
                                        className="navbar-toggle borderless collapsed">
                                        <DropdownButton
                                                  bsStyle="default"
                                                  className="glyphicon glyphicon-menu-hamburger"
                                                  id="hamburguer-menu"
                                                  noCaret={true}
                                                  pullRight={true}
                                                  title="">
                                                  <MenuItem
                                                            bsClass="dropdown"
                                                            disabled={false}
                                                            divider={false}
                                                            header={true}>
                                                            Categories
                                                  </MenuItem>
                                                  <MenuItem
                                                            bsClass="dropdown"
                                                            disabled={false}
                                                            divider={false}
                                                            eventKey={0}
                                                            header={false}
                                                            onClick={[Function]}>
                                                            nameMock
                                                  </MenuItem>
                                                  <MenuItem
                                                            bsClass="dropdown"
                                                            disabled={false}
                                                            divider={true}
                                                            header={false} />
                                                  <MenuItem
                                                            bsClass="dropdown"
                                                            disabled={false}
                                                            divider={false}
                                                            eventKey="4"
                                                            header={false}
                                                            onClick={[Function]}>
                                                            Contact Us
                                                  </MenuItem>
                                        </DropdownButton>
                              </div>
                              <a
                                        className="navbar-brand"
                                        href="home">
                                        Universal React
                              </a>
                    </div>
                    <div
                              className="navbar-collapse collapse">
                              <ul
                                        className="nav navbar-nav navbar-right">
                                        <li>
                                                  <DropdownButton
                                                            className="categories-dropdown"
                                                            id="categories"
                                                            noCaret={true}
                                                            title="Categories">
                                                            <MenuItem
                                                                      bsClass="dropdown"
                                                                      disabled={false}
                                                                      divider={false}
                                                                      eventKey={0}
                                                                      header={false}
                                                                      onClick={[Function]}>
                                                                      nameMock
                                                            </MenuItem>
                                                  </DropdownButton>
                                        </li>
                                        <li>
                                                  <a
                                                            href="contact">
                                                            Contact Us
                                                  </a>
                                        </li>
                              </ul>
                    </div>
          </div>
</nav>,
        "_debugID": 2,
        "_renderedOutput": <nav
          className="navbar navbar-default navbar-fixed-top">
          <div
                    className="container-fluid">
                    <div
                              className="navbar-header">
                              <div
                                        className="navbar-toggle borderless collapsed">
                                        <DropdownButton
                                                  bsStyle="default"
                                                  className="glyphicon glyphicon-menu-hamburger"
                                                  id="hamburguer-menu"
                                                  noCaret={true}
                                                  pullRight={true}
                                                  title="">
                                                  <MenuItem
                                                            bsClass="dropdown"
                                                            disabled={false}
                                                            divider={false}
                                                            header={true}>
                                                            Categories
                                                  </MenuItem>
                                                  <MenuItem
                                                            bsClass="dropdown"
                                                            disabled={false}
                                                            divider={false}
                                                            eventKey={0}
                                                            header={false}
                                                            onClick={[Function]}>
                                                            nameMock
                                                  </MenuItem>
                                                  <MenuItem
                                                            bsClass="dropdown"
                                                            disabled={false}
                                                            divider={true}
                                                            header={false} />
                                                  <MenuItem
                                                            bsClass="dropdown"
                                                            disabled={false}
                                                            divider={false}
                                                            eventKey="4"
                                                            header={false}
                                                            onClick={[Function]}>
                                                            Contact Us
                                                  </MenuItem>
                                        </DropdownButton>
                              </div>
                              <a
                                        className="navbar-brand"
                                        href="home">
                                        Universal React
                              </a>
                    </div>
                    <div
                              className="navbar-collapse collapse">
                              <ul
                                        className="nav navbar-nav navbar-right">
                                        <li>
                                                  <DropdownButton
                                                            className="categories-dropdown"
                                                            id="categories"
                                                            noCaret={true}
                                                            title="Categories">
                                                            <MenuItem
                                                                      bsClass="dropdown"
                                                                      disabled={false}
                                                                      divider={false}
                                                                      eventKey={0}
                                                                      header={false}
                                                                      onClick={[Function]}>
                                                                      nameMock
                                                            </MenuItem>
                                                  </DropdownButton>
                                        </li>
                                        <li>
                                                  <a
                                                            href="contact">
                                                            Contact Us
                                                  </a>
                                        </li>
                              </ul>
                    </div>
          </div>
</nav>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <NavHeader
    categories={
        Array [
            Object {
              "href": "hrefMock",
              "name": "nameMock",
            },
          ]
    } />,
}
`;
